src/ashley/models.py

```diff
@@ -197,3 +212,50 @@ class Meta:
 
 # Default implementation of the AbstractLTIContext model
 LTIContext = model_factory(AbstractLTIContext)
+
+
+def image_directory_path(instance, filename):
+    """File will be uploaded to specific path"""
+    return "{0}_{1}/{2}/{3}".format(
+        instance.forum.id, instance.forum.slug, instance.poster.id, filename
+    )
+
+
+class AbstractUploadImage(DatedModel):
+    """Abstract model for images."""
+
+    file = models.ImageField(
+        upload_to=image_directory_path,
+        max_length=100,
+        blank=True,
+        validators=[
+            validate_upload_image_file_size,
+            validate_upload_image_allowed_type,
+        ],
+        null=True,
+    )
+    forum = models.ForeignKey(
+        "forum.Forum",
+        on_delete=models.CASCADE,
+        verbose_name=_("Forum"),
+        blank=True,
+        null=True,
+    )
+    poster = models.ForeignKey(
+        get_user_model(),
+        blank=True,
+        null=True,
+        on_delete=models.CASCADE,
+        verbose_name=_("Poster"),
+    )
+
+    class Meta:
+        """Options for the `AbstractImage` model."""
+
+        abstract = True
+        app_label = "ashley"
+        ordering = ["-created", "id"]
```

Comment:
Can you explain in a code comment why this is necessary?

Reply:
In fact, necessary or not, I don't understand the purpose of this. That's why a comment would be great!

Reply:
no purpose i'll get ride of ti